src/TermInterface.jl

```diff
@@ -45,12 +45,41 @@ and pattern matching features.
 function exprhead end
 export exprhead
 
+"""
+  head(x)
+
+If `x` is a term as defined by `istree(x)`, `head(x)` returns the head of the
+term if `x`. The `head` type has to be provided by the package.
+"""
+function head end
+export head
+
+"""
+  head_symbol(x::HeadType)
+
+If `x` is a head object, `head_symbol(T, x)` returns a `Symbol` object that
+corresponds to `y.head` if `y` was the representation of the corresponding term
+as a Julia Expression. This is useful to define interoperability between
+symbolic term types defined in different packages and should be used when
+calling `maketerm`.
+"""
+function head_symbol end
+export head_symbol
```

Comment:
This should not be in this package. This package should never care about `Symbol`.

Reply:
It should be fine to move this to Metatheory.

Reply:
I agree this should not be in the package, it feels like exprhead and it doesn't make sense for many applications of this library